app/views/handlers/list_collector.py

```diff
@@ -79,12 +79,16 @@ def _is_list_collector_complete(self):
             for list_item_id in list_model.items)
 
     def _get_incomplete_repeating_block_ids(self) -> Generator:
-        list_model = self._questionnaire_store.list_store.get(self.rendered_block["for_list"])
         repeating_blocks = self.rendered_block.get("repeating_blocks")
+        if not repeating_blocks:
+            return
 
+        list_model = self._questionnaire_store.list_store.get(self.rendered_block["for_list"])
         for list_item_id in list_model.items:
-            if not self.questionnaire_store_updater.is_section_complete(self.current_location.section_id, list_item_id):
-                complete_block_ids = self.questionnaire_store_updater.get_completed_block_ids(self.current_location.section_id, list_item_id)
-                for repeating_block in repeating_blocks:
-                    if repeating_block["id"] not in complete_block_ids:
-                        yield list_item_id, repeating_block["id"]
+            if self.questionnaire_store_updater.is_section_complete(self.current_location.section_id, list_item_id):
+                continue
+
+            complete_block_ids = self.questionnaire_store_updater.get_completed_block_ids(self.current_location.section_id, list_item_id)
+            for repeating_block in repeating_blocks:
+                if repeating_block["id"] not in complete_block_ids:
```

Comment:
walrus the id?

Reply:
Don't think it's walrus-able after latest changes - but maybe I can't figure out the syntax